### PR TITLE
En_Syateki_Okuta (Shooting Gallery Octorok) OK

### PR DESCRIPTION
**Pull request by tom-overton**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

The last of the Syateki actors. Like all the previous ones, this one is undocumented; I *suppose* I could start doing that as part of this PR, but I'd rather do it separately.

### CODE REVIEW

File: src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c [hunk 1]
```diff
@@ -35,72 +41,482 @@ const ActorInit En_Syateki_Okuta_InitVars = {
...
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+s32 func_80A36A90(EnSyatekiOkuta* this, GlobalContext* globalCtx) {
+    if ((this->actionFunc == func_80A365EC) || (this->actionFunc == func_80A36350)) {
+        return false;
+    }
+
+    if ((this->collider.base.acFlags & AC_HIT) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c:332]:
```suggestion
    if (this->collider.base.acFlags & AC_HIT) {
```
Needed?

File: src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c [hunk 2]
```diff
@@ -35,72 +41,482 @@ const ActorInit En_Syateki_Okuta_InitVars = {
...
+
+    this->actionFunc(this, globalCtx);
+
+    if (this->actionFunc != func_80A36350) {
+        SkelAnime_Update(&this->skelAnime);
+    }
+
+    func_80A36AF8(this, globalCtx);
+
+    if (func_80A36A90(this, globalCtx) != 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c:369]:
```suggestion
    if (func_80A36A90(this, globalCtx)) {
```
looks to return a bool

File: src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c [hunk 3]
```diff
@@ -35,72 +41,482 @@ const ActorInit En_Syateki_Okuta_InitVars = {
...
+            arg2->x = arg2->y = 1.0f;
+        }
+    } else {
+        return false;
+    }
+
+    return true;
+}
+
+s32 func_80A37294(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Syateki_Okuta/z_en_syateki_okuta.c:466]: Looks to be the OverrideLimbDraw

**Review by hensldm — APPROVED** [review]
Just a few minor things.

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**